lib/ansible/modules/system/pids.py

```diff
@@ -52,8 +52,23 @@
     HAS_PSUTIL = False
 
 
+def compare_lower(a, b):
+    if a is None or b is None:
+        # this could just be "return False" but would lead to surprising behavior if both a and b are None
+        return a == b
+
+    return a.lower() == b.lower()
+
+
 def get_pid(name):
-    return [p.info['pid'] for p in psutil.process_iter(attrs=['pid', 'name']) if p.info and p.info.get('name', None) and p.info['name'].lower() == name.lower()]
+    pids = []
+
+    for proc in psutil.process_iter(attrs=['name', 'cmdline']):
```

Comment:
Shouldn't this be including 'pid' as per original code ?

Reply:
Only if you access the pid through the `p.info` dict attribute; the `Process` object itself [will always have a `pid` attribute set](https://github.com/giampaolo/psutil/blob/c10df5aa04e1ced58d19501fa42f08c1b909b83d/psutil/__init__.py#L534-L537).

Reply:
@Akasurde thoughts?

Reply:
@Akasurde ping

Reply:
@wmedlar Sorry for delay, I will take a look.